lib/src/polygon_clipping/vector.dart

```diff
@@ -0,0 +1,26 @@
+import 'dart:math';
+
+import 'package:turf/src/polygon_clipping/utils.dart';
+
+/* Cross Product of two vectors with first point at origin */
+num crossProduct(Point a, Point b) => a.x * b.y - a.y * b.x;
+
+/* Dot Product of two vectors with first point at origin */
+num dotProduct(Point a, Point b) => a.x * b.x + a.y * b.y;
+
+/* Comparator for two vectors with same starting point */
+num compareVectorAngles(Point basePt, Point endPt1, Point endPt2) {
+  double res = orient2d(
+    endPt1.x.toDouble(),
+    endPt1.y.toDouble(),
+    basePt.x.toDouble(),
+    basePt.y.toDouble(),
+    endPt2.x.toDouble(),
+    endPt2.y.toDouble(),
+  );
+  return res > 0
+      ? -1
+      : res < 0
+          ? 1
+          : 0;
```

Comment:
we already have implemented those functions on `Position`s

Reply:
I can't seem to find the equivalent of the compareVectorAngles. Do you happen to know which method you are referring to? I see the dot and cross product.

Reply:
ok, `compareVectorAngles` is the one function we don't have, but we have most of the others (esp. basic vector arithmetic like + - )

You can just use:
`Position(...) + Position(...)` – it uses operator overloading in dart